File: fib.cs
//------------------------------------------------------------------------------
// BigNum
// $ csc -o+ -w:4 -r:System.Numerics.dll -out:fib.cs.exe fib.cs
// $ fib.cs.exe [NUM]
//------------------------------------------------------------------------------
using System;
using System.IO;
using System.Numerics;
using System.Text;

namespace Program
{
	internal class Sub
	{
		private static void SubFib(int iNum = 0)
		{
			if (iNum < 0)
			{
				return;
			}

			Console.WriteLine("0\t0");
			if (iNum < 1)
			{
				return;
			}

			Console.WriteLine("1\t1");
			if (iNum < 2)
			{
				return;
			}

			BigInteger fib0 = 0;
			BigInteger fib1 = 1;
			BigInteger fib2 = 0;
			int iCnt = 1;

			StreamWriter SW = new StreamWriter(Console.OpenStandardOutput()) { AutoFlush = false };
			Console.SetOut(SW);
			StringBuilder Buf = new StringBuilder(10_000_000);

			while (iNum > iCnt)
			{
				fib2 = fib1 + fib0;
				fib0 = fib1;
				fib1 = fib2;
				++iCnt;

				_ = Buf.Append(iCnt.ToString());
				_ = Buf.Append("\t");
				_ = Buf.Append(fib1.ToString());
				_ = Buf.Append(Environment.NewLine);
			}

			Console.Write(Buf.ToString());
			Console.Out.Flush();
		}

		private static void Main(string[] ARGS)
		{
			int iNum = 0;
			if (ARGS.Length > 0)
			{
				int.TryParse(ARGS[0], out iNum);
			}
			SubFib(iNum);
		}
	}
}

File: fib.js
#!node
//---------------------------------------------------------------------
// BigNum
// $ node fib.js [NUM]
//---------------------------------------------------------------------
function subFib(
	num
)
{
	if(num < 0)
	{
		return;
	}

	const OS = require("os");
	NL = OS.EOL;

	process.stdout.write("0\t0" + NL);
	if(num < 1)
	{
		return;
	}

	process.stdout.write("1\t1" + NL);
	if(num < 2)
	{
		return;
	}

	let fib0 = BigInt(0);
	let fib1 = BigInt(1);
	let fib2 = BigInt(0);
	let cnt = 1;

	const BufMax = 4 * 1000;
	let Buf = [BufMax];
	let BufIndex = 0;

	while(num > cnt)
	{
		fib2 = fib1 + fib0;
		fib0 = fib1;
		fib1 = fib2;
		++cnt;

		Buf[BufIndex + 0] = cnt;
		Buf[BufIndex + 1] = "\t";
		Buf[BufIndex + 2] = fib1;
		Buf[BufIndex + 3] = NL;
		BufIndex += 4;

		if(BufIndex >= BufMax)
		{
			process.stdout.write(Buf.join(""));
			BufIndex = 0;
		}
	}

	process.stdout.write(Buf.slice(0, BufIndex).join(""));
}

function main()
{
	let num = 0;
	const ARGS = process.argv.slice(2);
	if(! isNaN(ARGS[0]))
	{
		num = parseInt(ARGS[0], 10);
	}
	subFib(num);
}

main();
